tests/core/util/test_streamable.py

```diff
@@ -373,6 +374,64 @@ def test_parse_str(self):
         with raises(AssertionError):
             parse_str(io.BytesIO(b"\x00\x00\x02\x01" + b"a" * 512))
 
+    def test_ignored_field(self):
+        @dataclass(frozen=True)
+        @streamable
+        class Bare(Streamable):
+            a: uint32
+            b: uint32
+            c: List[uint32]
+            d: List[List[uint32]]
+            e: Optional[uint32]
+            f: Optional[uint32]
+            g: Tuple[uint32, str, bytes]
+
+        bare = Bare(a=24, b=352, c=[1, 2, 4], d=[[1, 2, 3], [3, 4]], e=728, f=None, g=(383, "hello", b"goodbye"))
+
+        bare_bytes = bytes(bare)
+        assert bare == Bare.from_bytes(bare_bytes)
+
+        @dataclass(frozen=True)
+        @streamable
+        class Ignored(Streamable):
+            a: uint32
+            b: uint32
+            c: List[uint32]
+            d: List[List[uint32]]
+            e: Optional[uint32]
+            f: Optional[uint32]
+            g: Tuple[uint32, str, bytes]
+            m: uint32 = unstreamed_field(init=False)
```

Comment:
It seems like we're initializing `m` to a value that is not a `uint32`. I don't understand, can you explain? @altendky

Reply:
Ah, got it, it's an extension of the `dataclasses` magic.

> Scary and confusing? I thought the previous inconsistent use of `.__annotations__` was the scary part, as shown with the breakage in 3.10 (though I didn't look into what really broke). `:]` The caching of the header hash was the inspiration to look at this but it is very much a self-justified shift to a cleaner, more explicit, and more extensible design. Or so I thought...

Apologies for this comment... it's more me reacting to the fact that I don't recognize this file anymore, and it's gotten so complicated that I don't feel like I can get my head around it enough to properly review this patch without setting aside significant time to study it. I fully expect given your history that the work you've done is likely well-reasoned and thoughtful of trade-offs. I just don't understand it yet.